File: pr1-java/server/src/main/java/pr1Java/server/servers/RpcConcurrentServer.java
package pr1Java.server.servers;

import pr1Java.networking.rpcProtocol.ClientRpcProxy;
import pr1Java.server.Configuration;
import pr1Java.services.IServices;

import java.net.Socket;

public class RpcConcurrentServer extends ConcurrentServer {

    private final IServices services;

    public RpcConcurrentServer(Integer port, IServices services) {
        super(port);
        this.services = services;
    }

    @Override
    protected Thread createClientProxyThread(Socket connection) {
        Configuration.logger.traceEntry("entering with", connection);

        ClientRpcProxy clientProxy = new ClientRpcProxy(services, connection);
        Thread clientProxyThread = new Thread(clientProxy);

        Configuration.logger.traceExit(clientProxyThread);
        return clientProxyThread;
    }
}

File: pr1-cs/Server/Servers/RpcConcurrentServer.cs
using System.Net.Sockets;
using System.Threading;
using Networking;
using Services;

namespace Server.Servers
{
    public class RpcConcurrentServer : ConcurrentServer
    {
        private readonly IServices _services;
        
        public RpcConcurrentServer(string host, int port, IServices services) : base(host, port)
        {
            _services = services;
        }

        protected override Thread CreateClientProxyThread(TcpClient connection)
        {
            var clientProxy = new RpcClientProxy(_services, connection);
            return new Thread(clientProxy.Run);
        }
    }
}